distributed/worker.py

```diff
@@ -2149,7 +2287,8 @@ async def gather_dep(self, worker, dep, deps, total_nbytes, cause=None):
                         self.release_key(d)
                         continue
                     elif ts.state not in ("ready", "memory"):
-                        self.transition(ts, "waiting", worker=worker, remove=not busy)
+                        # "waiting" or "fetch"?
+                        self.transition(ts, "fetch", worker=worker, remove=not busy)
```

Comment:
afaiu it should be `fetch` but the `remove` toggle plays a crucial role here. It might no longer be necessary, in fact.

Reply:
yeah, I agree it's a `fetch` here -- I also tried removing the `remove` toggle in 2a8c0e2 and it looks (tentatively) that you're correct that it is no longer needed now that `waiting` and `fetch` are disambiguated.